include/swift/AST/DiagnosticsParse.def

```diff
@@ -862,6 +862,8 @@ ERROR(expected_lbrace_after_if,PointsToFirstBadToken,
       "expected '{' after 'if' condition", ())
 ERROR(expected_lbrace_after_else,PointsToFirstBadToken,
       "expected '{' after 'else'", ())
+ERROR(unexpected_else_after_if,PointsToFirstBadToken,
```

Comment:
`PointsToFirstBadToken` means: if `else` is at the new line, diagnose should point the end of the conditions.
```swift
if <condition>
//            ^ here
else { }
```
should be `none` in this case.